src/content/marina/marinaBroker.ts

```diff
@@ -178,10 +178,22 @@ export default class MarinaBroker extends Broker {
           const lbtc = lbtcAssetByNetwork(selectNetwork(state));
           const feeAsset = feeAssetHash ? feeAssetHash : lbtc;
 
+          // validate if fee asset is valid
           if (![lbtc, ...selectTaxiAssets(state)].includes(feeAsset)) {
             throw new Error(`${feeAsset} not supported as fee asset.`);
           }
 
+          // validate object recipient (asset and value)
+          // if no asset is present, assume lbtc for the current network
+          for (const recipient of recipients) {
+            if (!recipient.asset) {
+              if (!lbtc) throw new Error('missing asset on recipient');
+              recipient.asset = lbtc;
+            }
+            if (!recipient.value) throw new Error('missing value on recipient');
+            if (recipient.value < 0) throw new Error('negative value on recipient');
```

Comment:
worth checking floating point numbers as not valid

Reply:
Adding Number.isSafeInteger in 277de1f

But, `Number.isSafeInteger(1.0)` is true, which means a value of `1.0` would continue downstream.
Should I `recipient.value = parseInt(recipient.value, 10)` after this test?

Reply:
yes add extra parseInt indeed

Reply:
good catch